announcements.py

```diff
@@ -1,61 +1,92 @@
 import datetime
+import json
 import time
-# import requests
+import requests
+import itertools
 from multiprocessing import Process
 
-class MinsBeforeAnnouncement:
-    def __init__(self, minsBefore, text):
-        self.minsBefore = minsBefore
+class Announcement:
+    id_iter = itertools.count()
+    def __init__(self, time, text, banana_time):
+        self.id = next(self.id_iter)
+        self.time = time
         self.text = text
+        self.banana_time = banana_time
 
-class MinsBeforeAnnouncementWorker(Process):
-    def __init__(self, minsBefore, text, time):
-        super(MinsBeforeAnnouncementWorker, self).__init__()
-        self.minsBefore = minsBefore
-        self.text = text
+class AnnouncementWorker(Process):
+    def __init__(self, id, time, text, banana_time):
+        super(AnnouncementWorker, self).__init__()
+        self.id = id
         self.time = time
+        self.text = text
+        self.banana_time = banana_time
 
     def run(self):
         while True:
```

Comment:
Can avoid the concept of having an infinite loop by making the process terminate upon completion of its job. Might make more sense for the `manager` to create a new process upon completion for the next occurence.

Thoughts?

Reply:
Sounds like a good idea. I'm also not a fan of the infinite loop. I will create an issue for this however as it is a bit more than just a small change.

Reply:
Issue created: https://github.com/NickoOwen/bananabot/issues/6